pkg/skaffold/docker/parse_test.go

```diff
@@ -486,10 +486,21 @@ func TestGetDependencies(t *testing.T) {
 			expected:    []string{"Dockerfile", "file"},
 			fetched:     []string{"jboss/wildfly:14.0.1.Final"},
 		},
+		{
+			description: "build args with an environment variable",
+			dockerfile:  copyServerGoBuildArg,
+			workspace:   ".",
+			buildArgs:   map[string]*string{"FOO": util.StringPtr("{{.FILE_NAME}}")},
+			expected:    []string{"Dockerfile", "server.go"},
+			fetched:     []string{"ubuntu:14.04"},
+		},
 	}
 
 	for _, test := range tests {
 		t.Run(test.description, func(t *testing.T) {
+			util.OSEnviron = func() []string {
```

Comment:
please introduce env vars as a field on the testcases struct and set `FILEN_NAME=server.go` only for the "build args with an environment variable" testcase